src/rules/noShadowedVariableRule.ts

```diff
@@ -31,15 +31,37 @@ class NoShadowedVariableWalker extends Lint.BlockScopeAwareRuleWalker<ScopeInfo,
         return new ScopeInfo();
     }
 
+    public visitBindingElement(node: ts.BindingElement) {
+        // TODO: handle node.dotDotDotToken?
+        var isSingleVariable = node.name.kind === ts.SyntaxKind.Identifier;
```

Comment:
I think this is worth extracting into a utility method. I've seen it at least four times in your recent PRs.